### ISSUE TITLE
Federation health check

### ISSUE THREAD

**Issue by betatim** [jupyterhub/mybinder.org-deploy#1104]
The federation proxy should start checking https://gke.mybinder.org/health and https://ovh.mybinder.org/health instead of the `/versions` end point it queries right now. I think all it would take is to change the "health" URLs in https://github.com/jupyterhub/mybinder.org-deploy/blob/5228ca55e98231e239d1bca522a94853f546213e/mybinder/values.yaml#L420-L429

This means that this task can be done by someone who wants to learn a bit more about how the federation setup works.

**Comment by bitnik** [comment]
I want to learn how federation works before workshop next week. Can I work on this if anybody has not already started?

**Comment by betatim** [comment]
It would be super cool to have someone who also knows about this :)

However this issue should have been closed as we already switched the end points that are checked :-/

Thinking what else could be a good starting point for learning. The health endpoint now includes information about how many pods are running and a hub can optionally announce what capacity it has (the pod quota part in https://gke.mybinder.org/health). One thing to do is to extend the federation proxy https://github.com/jupyterhub/mybinder.org-deploy/blob/master/images/federation-redirect/app.py to take into account whether or not a hub is above its capacity when picking a target. We'd also have to configure a capacity for OVH, I'd suggest 50 to start with as that seems to work as maximum and sometimes we go above it (-> we can observe the limiting works).
